tools/generate.go

```diff
@@ -216,26 +188,48 @@ func parseSegs(sourcePath string) ([]*Seg, string) {
 
 func parseAndRenderSegs(sourcePath string) ([]*Seg, string) {
 	segs, filecontent := parseSegs(sourcePath)
-	lexer := whichLexer(sourcePath)
+	var goFile bool
 	for _, seg := range segs {
 		if seg.Docs != "" {
 			seg.DocsRendered = markdown(seg.Docs)
 		}
 		if seg.Code != "" {
-			seg.CodeRendered = cachedPygmentize(lexer, seg.Code)
+			seg.CodeRendered = cachedPygmentize(sourcePath, seg.Code)
 			// adding the content to the js code for copying to the clipboard
 			if strings.HasSuffix(sourcePath, ".go") {
+				goFile = true
 				seg.CodeForJs = strings.Trim(seg.Code, "\n") + "\n"
 			}
 		}
 	}
 	// we are only interested in the 'go' code to pass to play.golang.org
-	if lexer != "go" {
+	if !goFile {
 		filecontent = ""
 	}
 	return segs, filecontent
 }
 
+func chromaFormat(code, fileExtension string) string {
```

Comment:
It's a bit confusing that the second arg is called `fileExtension` when you're passing in the whole path. Can you please add a comment to this function and align the arg name with usage?